### ISSUE TITLE
Extend period concept schemes when using from_template

### ISSUE THREAD

**Issue by canwaf** [GSS-Cogs/csvcubed#682]

**Comment by robons** [comment]
I don't think we need to focus on doing this just when templates are used. I think we should attack this problem for anything which extends `sdmx-d:refPeriod`.

**Comment by canwaf** [comment]
If you try to do this with everything that extends `sdmx-d:refPeriod` you're now committing to discovering what another service holds on a given object, or inferring it from manipulating a URI. I believe that's a mistake unless you limit its concepts to `xsd:date(time)` literals.

**Comment by robons** [comment]
I don't think we are committed to that. We already treat anything which extends `sdmx-d:refPeriod` specially since we auto-generate special date-time code lists for them anyway.  

In an ideal world, we'd get the user to define dependencies (which themselves define dependencies) which would help us understand whether what they've referenced ultimately subclasses `sdmx-d:refPeriod` - but the LDR isn't there just yet ;)

**Comment by CharlesRendle** [comment]
Questions for @canwaf (from refinement session)

1. Are the start and end dates included or excluded in the range/period?
2. What the representation of this would look like in a csv? How does the user represent an arbitrary date-time period?  1 col for start date/time and one for end date/time or a single column containing one of those date-time+period strings?
